src/_pytest/config/__init__.py

```diff
@@ -1305,6 +1316,69 @@ def _getconftest_pathlist(
             values.append(relroot)
         return values
 
+    def _get_append_ini_values(self, name: str, ini_type) -> List[str]:
+        value = []
+        # append_ini is a list of "ini=value" options
+        # append all values if multiple values are set for same ini-name,
+        # e.g. -a foo=bar1 -a foo=bar2 will append 'bar1 bar2' to foo
+        for ini_config in self._append_ini:
+            try:
+                key, append_ini_value = ini_config.split("=", 1)
+            except ValueError:
+                raise UsageError(
+                    "-a/--append-ini expects option=value style (got: {!r}).".format(
+                        ini_config
+                    )
+                )
+            else:
+                if key == name:
+                    if ini_type in ["pathlist", "args"]:
+                        value.append(append_ini_value)
+                    elif ini_type == "linelist":
+                        value.append("\n{}".format(append_ini_value))
+                    else:
+                        self._warn_or_fail_if_strict(
+                            "append_ini option invalid for argument '{}' with type '{}'".format(
+                                name, ini_type
+                            )
+                        )
+        return value
+
+    def _resolve_ini_option(
+        self,
+        name: str,
+        default_value: str,
+        override_value: Optional[str],
+        append_values: List,
+        ini_type: Optional[str],
+    ) -> Union[List, str, None]:
+        value = self.inicfg.get(name, None)
```

Comment:
It could, but I think it makes sense to leave here. If we want to refactor anything I think we should move `_getini` out, since that will actually give value on its own. `_resolve_ini_option` is a helper for `_getini`

Reply:
> but I think it makes sense to leave here.

Any reason why? I think since `default_value`, `override_value`, and `append_value` are all passed in, it's inconsistent to fetch the `ini_value` inside. So I think it will be nicer to have it with the others, i.e. all four "components" (default, ini, override, append) are fetched, then passed to `_resolve_ini_option` to sort the final value out.

Reply:
Only because it's a helper to `_getini`, so it ends up depending on the class the way it's written right now.

But, I'm not opposed to moving it either. I'll move it when I push a commit responding to your feedback